server/app/views/admin/programs/ProgramIndexViewV2.java

```diff
@@ -0,0 +1,337 @@
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.each;
+import static j2html.TagCreator.h1;
+import static j2html.TagCreator.input;
+import static j2html.TagCreator.label;
+import static j2html.TagCreator.p;
+
+import auth.CiviFormProfile;
+import com.google.inject.Inject;
+import com.typesafe.config.Config;
+import controllers.admin.routes;
+import j2html.tags.ContainerTag;
+import j2html.tags.Tag;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.Optional;
+import java.util.concurrent.CompletionException;
+import play.mvc.Http;
+import play.twirl.api.Content;
+import services.LocalizedStrings;
+import services.program.ActiveAndDraftPrograms;
+import services.program.ProgramDefinition;
+import views.BaseHtmlView;
+import views.HtmlBundle;
+import views.admin.AdminLayout;
+import views.components.LinkElement;
+import views.components.Modal;
+import views.style.ReferenceClasses;
+import views.style.StyleUtils;
+import views.style.Styles;
+
+/** Renders a page so the admin can view all active programs and draft programs. */
+public final class ProgramIndexViewV2 extends BaseHtmlView {
+  private final AdminLayout layout;
+  private final String baseUrl;
+  private final ZoneId zoneId;
+
+  @Inject
+  public ProgramIndexViewV2(AdminLayout layout, Config config, ZoneId zoneId) {
+    this.layout = checkNotNull(layout);
+    this.baseUrl = checkNotNull(config).getString("base_url");
+    this.zoneId = checkNotNull(zoneId);
+  }
+
+  public Content render(
+      ActiveAndDraftPrograms programs, Http.Request request, Optional<CiviFormProfile> profile) {
+    if (profile.isPresent() && profile.get().isProgramAdmin() && !profile.get().isCiviFormAdmin()) {
+      layout.setOnlyProgramAdminType();
+    }
+
+    String pageTitle = "All programs";
+    ContainerTag publishAllModalContent =
+        div()
+            .withClasses(Styles.FLEX, Styles.FLEX_COL, Styles.GAP_4)
+            .with(p("Are you sure you want to publish all programs?").withClasses(Styles.P_2))
+            .with(maybeRenderPublishButton(programs, request));
+    Modal publishAllModal =
+        Modal.builder("publish-all-programs-modal", publishAllModalContent)
+            .setModalTitle("Confirmation")
+            .setTriggerButtonText("Publish all programs")
+            .build();
+
+    Tag contentDiv =
+        div()
+            .withClasses(Styles.PX_20)
+            .with(
+                h1(pageTitle).withClasses(Styles.MY_4),
+                div()
+                    .withClasses(Styles.FLEX, Styles.ITEMS_CENTER)
+                    .with(renderNewProgramButton())
+                    .with(div().withClass(Styles.FLEX_GROW))
+                    .condWith(programs.anyDraft(), publishAllModal.getButton()),
+                div()
+                    .withClasses(ReferenceClasses.ADMIN_PROGRAM_CARD_LIST, Styles.INVISIBLE)
+                    .with(
+                        p("Loading")
```

Comment:
suggestion (non-blocking): use i18n message rather than hard-coded string

Reply:
feel free to ignore

Reply:
Given that this is in the Admin UI where we have quite a few unlocalized messages, I'm going to leave as-is. Should we decide to localize the Admin UI, this would naturally fall out of that effort.

Reply:
I was thinking that it's a bit more generic, and if we're going to introduce a loading or spinner anywhere in the applicant UI it'd be good to have it translated already. Totally find to leave as-is though.